### PR TITLE
allow extensions to provide widgets for search panel

### PR DESCRIPTION
**Pull request by petrjasek**
SDCP-67

### CODE REVIEW

**Comment by tomaskikutis** [comment]
Can you explain how it works? Do interface `params` act simply as additional filters? I see that a component from extensions is getting rendered into the "advanced search" section of the search, but other than that I don't understand how an actual extension might work. It's not that there's something wrong with your code. I'm simply not familiar with search providers.

Can you go over how do custom media providers work in general? I assume it's not only a jpeg/mp4 files that we are getting from these providers, but also metadata. Do we have a transform function somewhere on the back-end which converts outside data structures to a Superdesk format?

**Comment by petrjasek** [comment]
yes the extensions provide additional filters based on external api capabilities, eg portrait/landspace pictures filter. params are sent to server together with query.
on the server there is a proxy for external api which sends request and then converts the results to superdesk like format, as you said.
usage: https://github.com/superdesk/superdesk-cp/pull/2

File: scripts/core/superdesk-api.d.ts [hunk 1]
```diff
@@ -77,9 +77,15 @@ declare module 'superdesk-api' {
                 };
             };
             iptcMapping?(data: IPTCMetadata, item: Partial<IArticle>): Promise<Partial<IArticle>>;
+            searchPanelWidgets?: Array<React.ComponentType<ISeachPanelWidgetProps>>;
         }
     }
 
+    export type ISearchPanelWidgetProps = {
+        provider: string;
+        params: any;
```
[the remark this answers is not in the document]
Comment (petrjasek) [reply scripts/core/superdesk-api.d.ts]: I don't see any common params. from/to is orangelogic specific

[the remark this answers is not in the document]
Comment (tomaskikutis) [reply scripts/core/superdesk-api.d.ts]: I see. Keep it generic then: `params: T;` and define from/to in the extension.

File: scripts/apps/search/components/search-panel-widgets.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ export class SearchPanelWidgets extends React.PureComponent<IProps> {
     render() {
         const widgets = flatMap(
             Object.values(extensions)
-                .map((extension) => extension.activationResult?.contributions?.searchPanelWidgets ?? [])
+                .map((extension) => extension.activationResult?.contributions?.searchPanelWidgets || [])
```
Comment (tomaskikutis) [inline scripts/apps/search/components/search-panel-widgets.tsx]: How is this a lint fix? Out of curiosity, I've checked it out the commit before this one and `npm run lint` passes without any errors.

Comment (petrjasek) [reply scripts/apps/search/components/search-panel-widgets.tsx]: npm run lint did pass, but not `grunt eslint:app` which we use on travis

Comment (tomaskikutis) [reply scripts/apps/search/components/search-panel-widgets.tsx]: Right. I now remember that I had the same issue myself and solved it by removing `eslint:app` task. Previously, 2 lint tasks were always being executed - `eslint:app` and `npm run lint`.

**Pull request merged**